models/Asset/Image/Thumbnail/Config/Dao.php

```diff
@@ -89,24 +92,50 @@ public function save()
             }
         }
 
-        $this->db->insertOrUpdate($data, $this->model->getName());
-        $this->autoClearTempFiles();
+        if($forceClearTempFiles) {
+            $this->db->insertOrUpdate($data, $this->model->getName());
+            $this->model->clearTempFiles();
+        } else {
+            $thumbnailDefinitionAlreadyExisted = $this->db->getById($this->model->getName()) !== null;
+
+            $this->db->insertOrUpdate($data, $this->model->getName());
+
+            if ($thumbnailDefinitionAlreadyExisted) {
+                $this->autoClearTempFiles();
+            }
+        }
+
+
     }
 
     /**
      * Deletes object from database
+     * @param bool $forceClearTempFiles force removing generated thumbnail files of saved thumbnail config
      */
-    public function delete()
+    public function delete($forceClearTempFiles = false)
     {
         $this->db->delete($this->model->getName());
-        $this->autoClearTempFiles();
+
+        if($forceClearTempFiles) {
+            $this->model->clearTempFiles();
+        } else {
+            $this->autoClearTempFiles();
+        }
     }
 
     protected function autoClearTempFiles()
     {
         $enabled = \Pimcore::getContainer()->getParameter('pimcore.config')['assets']['image']['thumbnails']['auto_clear_temp_files'];
         if ($enabled) {
-            $this->model->clearTempFiles();
+            $arguments = [
+                Console::getPhpCli(),
+                PIMCORE_PROJECT_ROOT.'/bin/console',
+                'pimcore:thumbnails:clear',
+                '--type=image',
+                '--name='.$this->model->getName()
+            ];
+
+            Console::execInBackground(implode(' ', $arguments));
```

Comment:
We should use `\Pimcore\Tool\Console::runPhpScriptInBackground()` instead, saves a bit of code and does proper escaping 😉